packages/babel-helpers/src/helpers/setFunctionName.ts

```diff
@@ -0,0 +1,22 @@
+/* @minVersion 7.23.5 */
+
+// https://tc39.es/ecma262/#sec-setfunctionname
+export default function setFunctionName(
```

Comment:
The helper is isolated because we will eventually emit `setFunctionName` when transforming the decorated anonymous class expression:

```js
// input.js
function dec(_, context) {
  // print '[Symbol.iterator]'
  console.log(context.name);
}
class Dummy {
  [Symbol.iterator] = @dec class {}
}
```

to

```js
function dec(_, context) {
  // print '[Symbol.iterator]'
  console.log(context.name);
}
var computedKey = Symbol.iterator;
class Dummy {
  [computedKey] = (setFunctionName(class {
    static {
      [_decorated_class, _initClass] = _applyDecs(this, [], [dec]).c;
    }
    static {
      _initClass();
    }
  }, computedKey), _decorated_class);
}
```
In this example we will also pass a symbol, which is also a valid property key, to the `setFunctionName` helper.

Currently we have not implemented such transform yet. It was addressed in #15122 but I will open new PR to do so.